packages/core/src/components/ProgressBar.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  defineComponent,
+  h,
+  inject,
+  onMounted,
+  onUpdated,
+  onUnmounted,
+  computed,
+} from '@vue/runtime-core'
+import type { ForegroundColor } from 'chalk'
+import type { PropType } from '@vue/runtime-core'
+import type { LiteralUnion } from '../utils'
+import { colorize } from '../renderer/textColor'
+import { scheduleUpdateSymbol } from '../injectionSymbols'
+
+const FIGURES = {
+  basic: '█',
+  shade: '▓',
+} as const
+
+type FigureType = keyof typeof FIGURES
+
+export const TuiProgressBar = defineComponent({
+  name: 'TuiProgressBar',
+
+  props: {
+    color: {
+      required: false,
+      default: 'blue',
+      type: String as PropType<LiteralUnion<ForegroundColor, string>>,
```

Comment:
I'm starting to think we should have a type named `ForegroundColorProp` somewhere 😆